### DISCUSSION TITLE
Installing SQLite builds LLVM from source

### DISCUSSION THREAD

**Discussion by MatthiasPortzel** [Homebrew/discussions#3974] [Everyday usage]
Installing SQLite builds LLVM from source, or at least attempts to. I just killed it after it ran in the background at full CPU for about an hour.

I don't have LLVM linked, so SQLite assumes it has to build the dependency from source. But running `brew link llvm` outputs "Warning: Refusing to link macOS provided/shadowed software: llvm."

I think this is a bug, but the bug template is very intense so I figured I'd start a discussion first to see if anyone else can reproduce.

I hit is updating from SQLite 3.39.x -> 3.40.0, and I can reproduce with `brew reinstall sqlite`.

**Comment by MatthiasPortzel** [comment]
As soon as I posted this the issue went away. `brew reinstall sqlite` now works fine.

<details>
<summary>My terminal log, to prove I'm not crazy.</summary>
<pre>
<code>
Running `brew update --auto-update`...
==> Auto-updated Homebrew!
Updated 3 taps (gromgit/fuse, homebrew/core and homebrew/cask).
==> New Casks
faxbot                               slab

You have 2 outdated formulae installed.
You can upgrade them with brew upgrade
or list them with brew outdated.

Using homebrew/cask
Using lagrange
Using emacs
Using aspell
Using vscodium
Using xquartz
Using racket
Using vlc
Using obs
Using transmission
Using maccy
Using alt-tab
Using macfuse
Using gromgit/fuse
Using gromgit/fuse/sshfs-mac
Using gromgit/fuse/ext4fuse-mac
Using coreutils
Using moreutils
Using less
Using tree
Using exa
Using ag
Using jq
Using ffmpeg
Using git
Using git-lfs
Using youtube-dl
Using rsync
Using openssl
Using qemu
Using nvm
Using deno
Using python@3.9
Using gcc@11
Using cmake
Using make
Using openjdk
Using openjdk@8
Using zig
Using ghc
Using elm
Using julia
Using ruby
Using meson
Using e2fsprogs
Using qemu
Upgrading bash
Using imagemagick
Using ninja
Using ccache
Using zstd
Using m4
Using autoconf
Using automake
Using libtool
Using libmagic
Using qt@5
Using amfora
Using micro
Using neovim
Using links
Using lynx
Using w3m
Using telnet
Using websocat
Using mitmproxy
Using btop
Homebrew Bundle complete! 67 Brewfile dependencies now installed.
==> Upgrading 1 outdated package:
sqlite 3.39.4 -> 3.40.0
==> Downloading https://ghcr.io/v2/homebrew/core/sqlite/manifests/3.40.0-
#=#=#                                                                    ########################################                                 ######################################################################## 100.0%
==> Downloading https://ghcr.io/v2/homebrew/core/sqlite/blobs/sha256:eae7
==> Downloading from https://pkg-containers.githubusercontent.com/ghcr1/b
######                                                                   ######################################################################## 100.0%
==> Upgrading sqlite
  3.39.4 -> 3.40.0

==> Pouring sqlite--3.40.0.monterey.bottle.1.tar.gz
🍺  /usr/local/Cellar/sqlite/3.40.0: 11 files, 4.4MB
==> Running `brew cleanup sqlite`...
Disable this behaviour by setting HOMEBREW_NO_INSTALL_CLEANUP.
Hide these hints with HOMEBREW_NO_ENV_HINTS (see `man brew`).
Removing: /usr/local/Cellar/sqlite/3.39.4... (11 files, 4.4MB)
Removing: /Users/matthias/Library/Caches/Homebrew/sqlite--3.39.4... (2MB)
==> No outdated dependents to upgrade!
==> Checking for dependents of upgraded formulae...
Disable this behaviour by setting HOMEBREW_NO_INSTALLED_DEPENDENTS_CHECK.
Hide these hints with HOMEBREW_NO_ENV_HINTS (see `man brew`).
==> Reinstalling 1 dependent with broken linkage from source:
llvm
==> Downloading https://ghcr.io/v2/homebrew/core/swig/manifests/4.1.0
#=#=#                                                                    ###################                                                      ######################################################################## 100.0%
==> Downloading https://ghcr.io/v2/homebrew/core/swig/blobs/sha256:dafaef
==> Downloading from https://pkg-containers.githubusercontent.com/ghcr1/b
########                                                                 ######################################################################## 100.0%
==> Downloading https://github.com/llvm/llvm-project/releases/download/ll
==> Downloading from https://objects.githubusercontent.com/github-product
                                                                         #                                                                        ##                                                                       ####                                                                     #####                                                                    #######                                                                  #########                                                                ############                                                             ##############                                                           ################                                                         ###################                                                      ######################                                                   #########################                                                ###########################                                              ##############################                                           ################################                                         ###################################                                      ######################################                                   ########################################                                 ###########################################                              ##############################################                           #################################################                        ####################################################                     #######################################################                  #########################################################                ############################################################             ###############################################################          ##################################################################       ####################################################################     #######################################################################  ######################################################################## 100.0%
==> Reinstalling llvm
==> Installing dependencies for llvm: swig
==> Installing llvm dependency: swig
==> Pouring swig--4.1.0.monterey.bottle.tar.gz
🍺  /usr/local/Cellar/swig/4.1.0: 775 files, 5.6MB
==> Installing llvm
==> cmake -G Unix Makefiles .. -DLLVM_ENABLE_PROJECTS=clang;clang-tools-e
==> cmake --build .
^C^C^C
One sec, cleaning up...
^C
One sec, cleaning up...
^C
One sec, cleaning up...

MacBookGamma:~ | matthias% brew doctor                                                                                                           20:24
Please note that these warnings are just used to help the Homebrew maintainers
with debugging if you file an issue. If everything you use Homebrew for is
working fine: please don't worry or file an issue; just ignore this. Thanks!

Warning: Unbrewed '.pc' files were found in /usr/local/lib/pkgconfig.
If you didn't put them there on purpose they could cause problems when
building Homebrew formulae, and may need to be deleted.

Unexpected '.pc' files:
  /usr/local/lib/pkgconfig/fuse-ext2.pc

Warning: You have unlinked kegs in your Cellar.
Leaving kegs unlinked can lead to build-trouble and cause formulae that depend on
those kegs to fail to run properly once built. Run `brew link` on these:
  six
  protobuf

Warning: Homebrew's "sbin" was not found in your PATH but you have installed
formulae that put executables in /usr/local/sbin.
Consider setting your PATH for example like so:
  echo 'export PATH="/usr/local/sbin:$PATH"' >> ~/.zshrc
MacBookGamma:~ | matthias% brew install llvm                                                                                                     20:25
Running `brew update --auto-update`...
==> Auto-updated Homebrew!
Updated 3 taps (gromgit/fuse, homebrew/core and homebrew/cask).

You have 1 outdated formula installed.
You can upgrade it with brew upgrade
or list it with brew outdated.

Warning: llvm 15.0.5 is already installed, it's just not linked.
To link this version, run:
  brew link llvm
MacBookGamma:~ | matthias% brew upgrade                                                                                                          20:25
==> Upgrading 1 outdated package:
libpng 1.6.38 -> 1.6.39
==> Downloading https://ghcr.io/v2/homebrew/core/libpng/manifests/1.6.39
######################################################################## 100.0%
==> Downloading https://ghcr.io/v2/homebrew/core/libpng/blobs/sha256:f7217880961411f37b5ba86376f5a6772bf45d1ae98ee86a01677439ce381cc0
==> Downloading from https://pkg-containers.githubusercontent.com/ghcr1/blobs/sha256:f7217880961411f37b5ba86376f5a6772bf45d1ae98ee86a01677439ce381cc0?s
######################################################################## 100.0%
==> Upgrading libpng
  1.6.38 -> 1.6.39

==> Pouring libpng--1.6.39.monterey.bottle.tar.gz
🍺  /usr/local/Cellar/libpng/1.6.39: 27 files, 1.3MB
==> Running `brew cleanup libpng`...
Disable this behaviour by setting HOMEBREW_NO_INSTALL_CLEANUP.
Hide these hints with HOMEBREW_NO_ENV_HINTS (see `man brew`).
Removing: /usr/local/Cellar/libpng/1.6.38... (27 files, 1.3MB)
MacBookGamma:~ | matthias% brew link llvm                                                                                                        20:26
Warning: Refusing to link macOS provided/shadowed software: llvm
If you need to have llvm first in your PATH, run:
  echo 'export PATH="/usr/local/opt/llvm/bin:$PATH"' >> ~/.zshrc

For compilers to find llvm you may need to set:
  export LDFLAGS="-L/usr/local/opt/llvm/lib"
  export CPPFLAGS="-I/usr/local/opt/llvm/include"
MacBookGamma:~ | matthias% brew install sqlite                                                                                                   20:26
Warning: sqlite 3.40.0 is already installed and up-to-date.
To reinstall 3.40.0, run:
  brew reinstall sqlite
MacBookGamma:~ | matthias% brew reinstall sqlite                                                                                                 20:26
==> Downloading https://ghcr.io/v2/homebrew/core/sqlite/manifests/3.40.0-1
Already downloaded: /Users/matthias/Library/Caches/Homebrew/downloads/97cc8d350a75175e595bb900b785ee6b825c155daca100e61a23579cadfcc112--sqlite-3.40.0-1.bottle_manifest.json
==> Downloading https://ghcr.io/v2/homebrew/core/sqlite/blobs/sha256:eae786ad782fb305cb1bfad22b37c52efc4c1f14614f29f7e3876dd048b83dd9
Already downloaded: /Users/matthias/Library/Caches/Homebrew/downloads/7927cfad56fcfb157ba8ed45be60953ccf0dbee54c6369af225ad99b2b735d5f--sqlite--3.40.0.monterey.bottle.1.tar.gz
==> Reinstalling sqlite
==> Pouring sqlite--3.40.0.monterey.bottle.1.tar.gz
🍺  /usr/local/Cellar/sqlite/3.40.0: 11 files, 4.4MB
==> Running `brew cleanup sqlite`...
Disable this behaviour by setting HOMEBREW_NO_INSTALL_CLEANUP.
Hide these hints with HOMEBREW_NO_ENV_HINTS (see `man brew`).
==> No outdated dependents to upgrade!
==> Checking for dependents of upgraded formulae...
Disable this behaviour by setting HOMEBREW_NO_INSTALLED_DEPENDENTS_CHECK.
Hide these hints with HOMEBREW_NO_ENV_HINTS (see `man brew`).
==> Reinstalling 1 dependent with broken linkage from source:
llvm
==> Downloading https://github.com/llvm/llvm-project/releases/download/llvmorg-15.0.5/llvm-project-15.0.5.src.tar.xz
Already downloaded: /Users/matthias/Library/Caches/Homebrew/downloads/1452e99170e8c58408b84c6bd77ec97f92800add5826c35f47f3747707de4147--llvm-project-15.0.5.src.tar.xz
==> Reinstalling llvm
^C
MacBookGamma:~ | matthias% brew update                                                                                                           20:27
Already up-to-date.
MacBookGamma:~ | matthias% brew link llvm                                                                                                        20:32
Error: No such keg: /usr/local/Cellar/llvm
MacBookGamma:~ | matthias% sqlite3                                                                                                               20:34
SQLite version 3.37.0 2021-12-09 01:34:53
Enter ".help" for usage hints.
Connected to a transient in-memory database.
Use ".open FILENAME" to reopen on a persistent database.
sqlite> ^D
MacBookGamma:~ | matthias% brew reinstall sqlite                                                                                                 20:37
==> Downloading https://ghcr.io/v2/homebrew/core/sqlite/manifests/3.40.0-1
Already downloaded: /Users/matthias/Library/Caches/Homebrew/downloads/97cc8d350a75175e595bb900b785ee6b825c155daca100e61a23579cadfcc112--sqlite-3.40.0-1.bottle_manifest.json
==> Downloading https://ghcr.io/v2/homebrew/core/sqlite/blobs/sha256:eae786ad782fb305cb1bfad22b37c52efc4c1f14614f29f7e3876dd048b83dd9
Already downloaded: /Users/matthias/Library/Caches/Homebrew/downloads/7927cfad56fcfb157ba8ed45be60953ccf0dbee54c6369af225ad99b2b735d5f--sqlite--3.40.0.monterey.bottle.1.tar.gz
==> Reinstalling sqlite
==> Pouring sqlite--3.40.0.monterey.bottle.1.tar.gz
🍺  /usr/local/Cellar/sqlite/3.40.0: 11 files, 4.4MB
==> Running `brew cleanup sqlite`...
Disable this behaviour by setting HOMEBREW_NO_INSTALL_CLEANUP.
Hide these hints with HOMEBREW_NO_ENV_HINTS (see `man brew`).
MacBookGamma:~ | matthias%
</code>
</pre>
</details>

**Reply by MatthiasPortzel** [reply]
`brew info llvm` now shows that it is now installed.

Although everything appears to be working, I'm concerned that I pressed ^C at non-optimal times and may have a broken system.

**Comment by carlocab** [comment]
What is the output of
```
brew linkage llvm
```
?
